### PR TITLE
ActivityLog: added user messages to ActivityLogMessageParser

### PR DESCRIPTION
**Pull request by fefrain**

### CODE REVIEW

File: server/modules/user/manager/userManager.js [hunk 1]
```diff
@@ -90,12 +93,17 @@ const updateUser = _userFetcher(_updateUser)
...
+    const userToRemove = await UserRepository.fetchUserByUuid(userUuidToRemove, t)
+    const logContent = {
+      [ActivityLog.keysContent.uuid]: userUuidToRemove,
+      [ActivityLog.keysContent.userName]: User.getName(userToRemove)
+    }
+    return await Promise.all([
+      AuthGroupRepository.deleteUserGroup(surveyId, userUuidToRemove, t),
+      ActivityLogRepository.insert(user, surveyId, ActivityLog.type.userRemove, logContent, false, t)
+    ])
+  })
```
Comment (minotogna) [inline server/modules/user/manager/userManager.js]: Users are never deleted from database. In the log I would save the userUuid to remove only, and in the fetch I would get the username

File: server/modules/activityLog/manager/activityLogManager.js [hunk 2]
```diff
@@ -81,9 +82,40 @@ const _getAvailableActivityTypes = async (surveyId, user) => {
...
+  const activityLogsDb = await ActivityLogRepository.fetch(surveyId, activityTypes, offset, limit)
+
+  return await Promise.all(activityLogsDb.map(async activityLogDb => {
+    if (R.includes(ActivityLog.getType(activityLogDb), [ActivityLog.type.userInvite, ActivityLog.type.userUpdate])) {
+      return await _transformActivityLogUser(surveyUuid)(activityLogDb)
+    } else {
+      return activityLogDb
+    }
+  }))
 }
```
Comment (minotogna) [inline server/modules/activityLog/manager/activityLogManager.js]: can't you get this with a join with user, auth_group and auth_group_user tables?

File: server/modules/user/manager/userManager.js (hunk 1 above)
Comment (fefrain) [reply server/modules/user/manager/userManager.js]: ok

File: server/modules/activityLog/manager/activityLogManager.js (hunk 2 above)
Comment (fefrain) [reply server/modules/activityLog/manager/activityLogManager.js]: yes, right

File: server/modules/activityLog/repository/activityLogRepository.js [hunk 3]
```diff
@@ -40,13 +40,32 @@ export const fetch = async (surveyId, activityTypes = null, offset = 0, limit =
...
+          public.auth_group_user agu
+        ON
+          agu.user_uuid = user_target.uuid 
+        JOIN  
+          public.auth_group ag
+        ON 
+          ag.uuid = agu.group_uuid AND ag.survey_uuid = s.uuid
+        WHERE 
+          s.id = $1
+      ) as target_user_removed
```
Comment (minotogna) [inline server/modules/activityLog/repository/activityLogRepository.js]: would selecting the target_user_uuid with left outer joins be more efficient?

Comment (fefrain) [reply server/modules/activityLog/repository/activityLogRepository.js]: yes, I will change it

File: server/modules/activityLog/repository/activityLogRepository.js [hunk 4]
```diff
@@ -23,8 +25,12 @@ export const insertMany = async (user, surveyId, activities, client) =>
   ])
 
 //===== READ
-export const fetch = async (surveyId, activityTypes = null, offset = 0, limit = 30, client = db) =>
-  await client.map(`
+export const fetch = async (surveyInfo, activityTypes = null, offset = 0, limit = 30, client = db) => {
+  const surveyUuid = Survey.getUuid(surveyInfo)
+  const surveyId = Survey.getIdSurveyInfo(surveyInfo)
+  const schema = getSurveyDBSchema(surveyId)
```
Comment (minotogna) [inline server/modules/activityLog/repository/activityLogRepository.js]: can't you pass just the surveyId as it was before? to get the surveyUuid you can join with survey table? it's just weird to pass the surveyInfo to the repository... isn't it?

Comment (fefrain) [reply server/modules/activityLog/repository/activityLogRepository.js]: you have always the surveyInfo when you call ActiviityLogRepository.fetch, so you can save a join, but I'll change it if you want

**Pull request merged**